config/history.go

```diff
@@ -220,35 +248,26 @@ func AddClipboardItem(text, fp string) error {
 	data.ClipboardHistory = append([]ClipboardItem{item}, data.ClipboardHistory...)
 
 	if len(data.ClipboardHistory) > ClipseConfig.MaxHistory {
-		for i := len(data.ClipboardHistory) - 1; i >= 0; i-- { // remove the first unpinned entry starting with the oldest
+		for i := len(data.ClipboardHistory) - 1; i >= 0; i-- {
+			// remove the first unpinned entry starting with the oldest
 			if !data.ClipboardHistory[i].Pinned {
 				data.ClipboardHistory = append(data.ClipboardHistory[:i], data.ClipboardHistory[i+1:]...)
 				break
 			}
 		}
 	}
 
-	if err = saveDataToFile(data); err != nil {
+	if err := WriteUpdate(data); err != nil {
 		return err
 	}
 	return nil
 }
 
 // This pins and unpins an item in the clipboard
 func TogglePinClipboardItem(timeStamp string) (bool, error) {
-	var data ClipboardHistory
+	var data ClipboardHistory = fileContents()
```

Comment:
I would say choose your team


```suggestion
	 data := fileContents()
```

Or


```suggestion
	var data ClipboardHistory
data = fileContents()
```